apis/logging/v1/output_types.go

```diff
@@ -225,3 +228,24 @@ type Loki struct {
 	// +optional
 	LabelKeys []string `json:"labelKeys,omitempty"`
 }
+
+// Splunk Deliver log data to Splunk’s HTTP Event Collector
+// Provides optional extra properties for `type: splunk_hec` ('splunk_hec_logs' after Vector 0.23
+type Splunk struct {
+	// The name of the index where to send the events to. If not specified, the default index is used.
+	// Note: This parameter supports Vector's template syntax, which enables you to use dynamic per-event values.
+	//
+	// +optional
+	Index string `json:"index,omitempty"`
+
+	// Fields to be added to Splunk index. https://docs.splunk.com/Documentation/Splunk/8.0.0/Data/IFXandHEC
+	//
+	// +optional
+	IndexedFields []string `json:"indexed_fields"`
+
+	// The sourcetype of events sent to this sink. If unset, Splunk will default to httpevent.
+	// Note: This parameter supports Vector's template syntax, which enables you to use dynamic per-event values.
+	//
+	// +optional
+	SourceType string `json:"sourcetype,omitempty"`
```

Comment:
@alanconway I believe we want to remove this based upon https://docs.splunk.com/Documentation/Splunk/8.0.0/Data/Listofpretrainedsourcetypes which defines a known format. Ours will always be JSON so my expectation is we may have some integration issues if we allow it to be set

Reply:
+1 we don't want to expose this now. We may offer other formats in future but we'll want to retain control of the splunk setting internally, not give it to the user.

Reply:
removed